uberfire-structure/uberfire-structure-backend/src/main/java/org/guvnor/structure/backend/pom/PomEditorDefault.java

```diff
@@ -0,0 +1,117 @@
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.apache.maven.model.io.xpp3.MavenXpp3Writer;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.guvnor.structure.pom.DynamicPomDependency;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.java.nio.file.Files;
+import org.uberfire.java.nio.file.Paths;
+import org.uberfire.java.nio.file.StandardOpenOption;
+
+public class PomEditorDefault implements PomEditor {
+
+    private final Logger logger = LoggerFactory.getLogger(PomEditorDefault.class);
+    private MavenXpp3Reader reader;
+    private MavenXpp3Writer writer;
+    private static final String DELIMITER = ":";
+
+
+    public PomEditorDefault() {
+        reader = new MavenXpp3Reader();
+        writer = new MavenXpp3Writer();
+    }
+
+    public boolean addDependency(DynamicPomDependency dep,
+                                 Path pomPath) {
+        if (dep.getGroupID().isEmpty() || dep.getArtifactID().isEmpty()) {
+            return false;
+        }
+
+        try {
+            Dependency pomDep = getMavenDependency(dep);
+            pomDep.setType("jar");
+            org.uberfire.java.nio.file.Path filePath = Paths.get(pomPath.toURI());
+            Model model = getPOMModel(filePath);
+            Set keys = getKeysFromDeps(model.getDependencies());
+            if(!keys.contains(getKeyFromDep(dep))) {
+                model.getDependencies().add(pomDep);
+            }else{
+                return false;
+            }
+            writePOMModelOnFS(filePath,
+                              model);
+        } catch (Exception ex) {
+            logger.error(ex.getMessage(),
+                         ex);
+            return false;
+        }
+        return true;
+    }
+
+    private String getKeyFromDep(DynamicPomDependency dep){
+        StringBuilder sb = new StringBuilder();
+        sb.append(dep.getGroupID()).append(DELIMITER).append(dep.getArtifactID()).append(DELIMITER).append(dep.getVersion()).toString();
+        return sb.toString();
+    }
+
+    private Set<String> getKeysFromDeps(List<Dependency> deps){
+        Set<String> depsMap = new HashSet<>(deps.size());
+        for(Dependency dep : deps){
+            StringBuilder sb = new StringBuilder();
+            sb.append(dep.getGroupId()).append(DELIMITER).append(dep.getArtifactId()).append(DELIMITER).append(dep.getVersion());
+            depsMap.add(sb.toString());
+        }
+        return depsMap;
+    }
+
+    private void writePOMModelOnFS(org.uberfire.java.nio.file.Path filePath,
+                                   Model model) throws IOException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
@desmax74 I don't know you and @porcelli are manipulating but isn't better to use our VFS api instead of manipulating the file directly? 

i.e. maybe we want to listen to this change to trigger a build.